docs/guides/nrfconnect_factory_data_configuration.md

```diff
@@ -0,0 +1,291 @@
+
+To check JSON file using JSON Schema manually on Linux machine install *php-json-schema* package:
+```
+$ sudo apt install php-json-schema
+```
+
+And use following command:
+```
+$ validate-json <path_to_JSON_file> <path_to_schema_file>
+```
+
+If in the console there are not any outputs, it means that JSON is correct.
+
+JSON file can be checked automatically by the python script during its generation by given a path to JSON schema file as an additional argument:
+```
+$ python generate_nrfconnect_chip_factory_data.py --schema <path_to_schema>
+```
+
+> Note: This [website](https://json-schema.org/understanding-json-schema/) can help better understand JSON schema. 
+
+### Preparing factory data partition on a device
+
+The factory data partition is an area in a device's persistent storage where a factory data set is stored. This area is configured using the [Partition Manager](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/scripts/partition_manager/partition_manager.html), within which all partitions are declared by writing the `pm_static.yaml` file. 
+
+Partitions and their sizes may be different according to different nRFconnect boards. Each nrfconnect board that the example supports contains a `pm_static.yaml` file in "*connectedhomeip/examples/< example_name >/nrfconnect/configuration/< pcb name >*" directory. 
+
+To prepare an example to support a factory data pm_static file has to contain a partition named `factory_data`, and Its size should be equal to at least 4 kB and be multiple of one flash page (for nrf52 and nrf53 boards, a single page is 4 kB length).
+
+Below, there is an example of how to create a factory data partition in pm_static.yaml file according to pm_static.yaml from [lock app example](../../examples/lock-app/nrfconnect/configuration/nrf52840dk_nrf52840/pm_static_dfu.yml) and nRF52840DK board:
+
+```
+...
+mcuboot_primary_app:
+    orig_span: &id002
+        - app
+    span: *id002
+    address: 0x7200
+    size: 0xf3e00
+
+factory_data:
+    address: 0xfb00
+    size: 0x1000
+    region: flash_primary
+
+settings_storage:
+    address: 0xfc000
+    size: 0x4000
+    region: flash_primary
+...
+```
+In this example, a `factory_data` partition has been placed after the application partition(mcuboot_primary_app is a container to store application firmware which can be replaced in the Direct Firmware Update process), and its size has been set to 4 kB (0x1000) which corresponds to one page of nrf52840 flash memory.
+Check a partition allocation using the `west` build system to ensure that the partition was appropriately created. Below there is a partition manager report output generated according to an example:
```

Comment:
Please add information that this command should be invoked in the example's directory.